liquidctl/driver/commander_pro.py

```diff
@@ -466,6 +468,12 @@ def set_color(self, channel, mode_str, colors, direction='forward', speed='mediu
         saved_effects = [] if mode_str == 'off' else self._data.load('saved_effects', default=[])
         saved_effects += [lighting_effect]
 
+        # check to make sure that too many LED effects are not being sent.
+        # the max seems to be 8 as found here https://github.com/liquidctl/liquidctl/issues/154#issuecomment-762372583
+        if len(saved_effects) > 8:
+            _LOGGER.warning(f'too many lighting effects. Run `liquidctl set {channel} color clear` to reset the effect')
+            return
```

Comment:
Shouldn't we maybe implement something like a ring buffer or some predetermined slots, instead of requiring user intervention?

After all, this is at least in part *our* abstraction, right?

Reply:
I thought about that, but I feel like it might lead to some confusion if previously sent (ie the first) lighting effects  just stop being applied when more are sent. And because the order that the effects are applied matters I felt this ensures that the user knows they have to rethink their effect if they hit this limit.

Reply:
Fair enough.